reactor-netty-http-brave/src/main/java/reactor/netty/http/brave/TracingHttpServerDecorator.java

```diff
@@ -266,27 +266,25 @@ static final class TracingMapHandle implements BiFunction<Mono<Void>, Connection
 		public Mono<Void> apply(Mono<Void> voidMono, Connection connection) {
 			HttpServerRequest braveRequest = connection.channel().attr(REQUEST_ATTR_KEY).get();
 			Span span = connection.channel().attr(SPAN_ATTR_KEY).get();
+			connection.onTerminate()
+			          .subscribe(
+			              null,
+			              t -> cleanup(connection.channel()),
+			              () -> cleanup(connection.channel()));
+			EventLoop eventLoop = connection.channel().eventLoop();
 			return voidMono.doFinally(sig -> {
 			                   if (braveRequest.unwrap() instanceof reactor.netty.http.server.HttpServerResponse) {
 			                       reactor.netty.http.server.HttpServerResponse response =
 			                               (reactor.netty.http.server.HttpServerResponse) braveRequest.unwrap();
 			                       Span localSpan = sig == SignalType.CANCEL ? span.annotate("cancel") : span;
 			                       HttpServerResponse braveResponse =
 			                               new DelegatingHttpResponse(response, braveRequest, throwable);
-			                       response.withConnection(conn -> {
-			                               conn.onTerminate()
-			                                   .subscribe(
-			                                           null,
-			                                           t -> cleanup(connection.channel()),
-			                                           () -> cleanup(connection.channel()));
-			                               EventLoop eventLoop = conn.channel().eventLoop();
-			                               if (eventLoop.inEventLoop()) {
-			                                   handler.handleSend(braveResponse, localSpan);
-			                               }
-			                               else {
-			                                   eventLoop.execute(() -> handler.handleSend(braveResponse, localSpan));
-			                               }
-			                       });
+			                       if (eventLoop.inEventLoop()) {
+			                           handler.handleSend(braveResponse, localSpan);
+			                       }
+			                       else {
+			                           eventLoop.execute(() -> handler.handleSend(braveResponse, localSpan));
+			                       }
```

Comment:
Took me a while to understand what's going on here, but I understand that at this point the `response.withConnection `lambda can potentially have the `DisposedConnection` which means there's no event loop associated any more so you are essentially capturing it in the closure through the above reference, correct?

Reply:
correct